library/hls/src/main/java/com/google/android/exoplayer2/source/hls/playlist/HlsPlaylistParser.java

```diff
@@ -368,6 +330,16 @@ private static HlsMasterPlaylist parseMasterPlaylist(LineIterator iterator, Stri
         variantInfosForUrl.add(
             new VariantInfo(
                 bitrate, videoGroupId, audioGroupId, subtitlesGroupId, closedCaptionsGroupId));
+      } else if (line.startsWith(TAG_STREAM_IFRAME)) {
+        String formatId = "iFrame-" + iFrameVariants.size();
+        int bitrate = parseStreamBitrate(variableDefinitions, line);
+        Format format = parseVideoContainerFormat(variableDefinitions, line, formatId, bitrate,
+            C.ROLE_FLAG_ALTERNATE | C.ROLE_FLAG_TRICK_PLAY);
+
```

Comment:
ROLE_FLAG_ALTERNATE should probably go, I added this to surface it to the track selector UI.